modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/GridDhtTxFinishFuture.java

```diff
@@ -435,8 +435,6 @@ private boolean finish(boolean commit,
         if (tx.onePhaseCommit())
             return false;
 
-        assert !commit || !tx.txState().mvccEnabled() || tx.mvccSnapshot() != null || F.isEmpty(tx.writeEntries());
```

Comment:
Do we need 'assert !commit || F.isEmpty(tx.writeEntries());' ?

Reply:
No, because `!tx.txState().mvccEnabled()` is a true now